src/addmult.c

```diff
@@ -336,57 +330,62 @@ void add_mult_line(char *line) {
 int init_and_load_multipliers(void) {
 
     FILE *cfp;
-    char s_inputbuffer[2000] = "";
+    char *s_inputbuffer = NULL;
+    size_t s_inputbuffer_len = 0;
     char *mults_location;
+    ssize_t read;
 
     if (mults_possible) {
-	/* free old array if exists */
-	g_ptr_array_free(mults_possible, TRUE);
+        /* free old array if exists */
+        g_ptr_array_free(mults_possible, TRUE);
     }
     mults_possible = g_ptr_array_new_with_free_func(free_possible_mult);
 
-
     if (strlen(multsfile) == 0) {
-	return 0;
+	    return 0;
     }
 
     mults_location = find_available(multsfile);
 
     if ((cfp = fopen(mults_location, "r")) == NULL) {
-	mvprintw(9, 0, "Error opening multiplier file %s.\n", multsfile);
-	refreshp();
-	sleep(5);
+        mvprintw(9, 0, "Error opening multiplier file %s.\n", multsfile);
+        refreshp();
+        sleep(5);
     }
 
     g_free(mults_location);
 
     if (cfp == NULL) {
-	return 0;       // couldn't open file
+	    return 0;       // couldn't open file
     }
 
-    while (fgets(s_inputbuffer, sizeof(s_inputbuffer), cfp) != NULL) {
-
-	/* strip leading and trailing whitespace */
-	g_strstrip(s_inputbuffer);
-
-	/* drop comments starting with '#' and empty lines */
-	if (*s_inputbuffer == '#' || *s_inputbuffer == '\0') {
-	    continue;
-	}
-
-	add_mult_line(s_inputbuffer);
-
+    while ((read = getline(&s_inputbuffer, &s_inputbuffer_len, cfp)) != -1) {
+            if (s_inputbuffer_len > 0) {
+                /* strip leading and trailing whitespace */
+                g_strstrip(s_inputbuffer);
+
+                /* drop comments starting with '#' and empty lines */
+                if (*s_inputbuffer == '#' || *s_inputbuffer == '\0') {
+                    continue;
+                }
+
+                add_mult_line(s_inputbuffer);
+            }
+            else {
+                perror("RuntimeError: ");
```

Comment:
This (and the other uses of `perror()`) is not too informative. If the (un)expected error occurs, you will get a message:

```
RuntimeError: 
```

You should consider to use `fprintf(stderr, "RuntimeError: %s:%ld", __FILE__, __LINE__)`

Reply:
According to documentation message will be: for code example:
`int main()
{
    perror( NULL );
    FILE *fp = fopen( "nonexisting.file", "rb" );
    if( !fp )
         perror( "RuntimeError: " );
    else
         fclose( fp );
   
    return 0;
}`
 "RuntimeError: No such file or directory" but if fprintf is a standard I can change it. Eventually add__FILE__ and __LINE__ to perror message.

Reply:
> According to documentation message will be: for code example:

Oh, that's new for me - thanks.

Then that would be fine for me.

> but if fprintf is a standard I can change it. 

I don't believe there is a "standard" way. In the (Tlf's) code above you check the result of memory allocation. Theoretically you will get an error message something like "Not enough memory" - without any context (eg. in which file and which function triggered the error). This is why I prefer the `fprintf()` with that way, but I do not consider it as "standard".

As I wrote, this would be fine for me. (With one note: this string is enough for argument: "RuntimeError" - no need the `:` and extra space)